### PR TITLE
Add basic code quality tooling 

### PR DESCRIPTION
**Pull request by fabiankrol**
- [x] credo 
- [x] ~~dogma~~ TIL: [mix format](https://hexdocs.pm/mix/master/Mix.Tasks.Format.html)

### CODE REVIEW

**Comment by tomekowal** [comment]
For mix format to work, just add `mix format --check-formatted` in the CI commands.
In our current project we also have a pre-commit hoook:
```
MIX_ENV=test mix do compile --warnings-as-errors
mix test
mix credo --strict
```
(we haven't enforced formatting yet in the project but for new project it is a good idea).
I am approving that PR. We can add `mix format` in a separate one.

**Comment by tomekowal** [comment]
One more thing. Dialyzer takes a lot more time on the first run then on subsequent runs. It is a good candidate for caching. The problem is I am not sure if it automatically detects when it should update `.plt` files. This requires a little bit of playing around to set up.

File: lib/zmq_ex.ex [hunk 1]
```diff
@@ -79,14 +75,13 @@ defmodule ZmqEx do
 
   def recv(socket, send?) do
     {:ok, msg} = :gen_tcp.recv(socket, 0)
-    IO.inspect(msg)
     dmsg = decode(msg)
-    IO.inspect(dmsg)
+    IO.puts(dmsg)
```
Comment (Dreat) [inline lib/zmq_ex.ex]: with printer_loop we don't need to print anything in recv

Comment (fabiankrol) [reply lib/zmq_ex.ex]: Sure, it makes sense to clean it up. But it's not a main goal of this pr

**Pull request merged**